BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
```

Comment:
This `Rate` type is shared between CNB response contract and our exchange rate api response contract. A change made for one could implicitly affect the other.
- Let's create a separate `ExchangeRate` type for our exchange rate api response contract
- Let's rename `Rate` class to `CnbRate` to reflect its origin
- Both should be of type `class` or `record` and not a `struct`. See [Choosing Between Class and Struct](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct)

```suggestion
public class CnbRate
```